pylint/checkers/base/name_checker/checker.py

```diff
@@ -513,18 +513,15 @@ def visit_assignname(  # pylint: disable=too-many-branches,too-many-statements
                 else:
                     node_type = "variable"
                     iattrs = tuple(node.frame().igetattr(node.name))
+                    attrs = tuple(node.frame().getattr(node.name))
```

Comment:
For the curious, we have to do the line number analysis on attrs. If we do it on iattrs, we're jumping out of the if/else into what other line defines the object that we inferred (`(1, 2, 3)` in the test case).